jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionCache.java

```diff
@@ -343,113 +341,60 @@ public Session get(String id) throws Exception
     protected Session getAndEnter(String id, boolean enter) throws Exception
     {
         Session session = null;
-        Exception ex = null;
 
-        while (true)
-        {
-            session = doGet(id);
+        session = doGet(id);
```

Comment:
You should just do computeIfAbsent here as it has an implicit doGet and act if null already